### ISSUE TITLE
NameError with wrapped setup.py (pdm.installers._editable_install)

### ISSUE THREAD

**Issue by blueyed** [pdm-project/pdm#407]
- [x] I have searched the issue tracker and believe that this is not a duplicate.

When trying to add mkdocs in editable mode it fails with a NameError due to `re` not being known when `exec`ing the code in `pdm.installers._editable_install`: https://github.com/pdm-project/pdm/blob/6baf013c4a83685148c649c0ca485397c3393048/pdm/installers/_editable_install.py#L11-L26

```
% .venv/bin/pdm add -e ~/Vcs/mkdocs -s doc -d
Adding packages to doc dev-dependencies: mkdocs
✔ 🔒 Lock successful
Changes are written to pdm.lock.
Changes are written to pyproject.toml.
Synchronizing working set with lock file: 18 to add, 0 to update, 0 to remove

  ✖ Install mkdocs 1.2.dev1 failed
  See /tmp/pdm-install-2rcus8kw.log for detailed debug log.
  [BuildError]: Call command ['/usr/bin/python', '-u', '/tmp/pdm-test/.venv/lib/python3.9/site-packages/pdm/installers/_editable_install.py', '/home/daniel/Vcs/mkdocs/setup.py', '/tmp/pdm-test/__pypackages__/3.9', '/tmp/pdm-test/__pypackages__/3.9/lib', '/tmp/pdm-test/__pypackages__/3.9/bin'] return non-zero status.
  Add '-v' to see the detailed traceback
.venv/bin/pdm add -e ~/Vcs/mkdocs -s doc -d  16.46s user 1.49s system 41% cpu 43.665 total
```

Note: this uses the `1.1.2` tag of mkdocs, with pdb 1.5.1.

Log:
```
Preparing isolated env for PEP 517 build...
running egg_info
writing mkdocs.egg-info/PKG-INFO
writing dependency_links to mkdocs.egg-info/dependency_links.txt
writing entry points to mkdocs.egg-info/entry_points.txt
writing requirements to mkdocs.egg-info/requires.txt
writing top-level names to mkdocs.egg-info/top_level.txt
warning: the 'license_file' option is deprecated, use 'license_files' instead
adding license file 'LICENSE' (matched pattern 'LICENSE')
reading manifest file 'mkdocs.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
warning: no files found matching '*.mustache' under directory 'mkdocs'
warning: no previously-included files matching '__pycache__' found under directory '*'
warning: no previously-included files matching '*.py[co]' found under directory '*'
writing manifest file 'mkdocs.egg-info/SOURCES.txt'
Preparing isolated env for PEP 517 build...
Traceback (most recent call last):
  File "/tmp/pdm-test/.venv/lib/python3.9/site-packages/pdm/installers/_editable_install.py", line 31, in <module>
    install(setup_py, prefix, lib_dir, bin_dir)
  File "/tmp/pdm-test/.venv/lib/python3.9/site-packages/pdm/installers/_editable_install.py", line 26, in install
    exec(compile(code, __file__, "exec"))
  File "…/Vcs/mkdocs/setup.py", line 43, in <module>
    version=get_version("mkdocs"),
  File "…/Vcs/mkdocs/setup.py", line 16, in get_version
    return re.search("__version__ = ['\"]([^'\"]+)['\"]", init_py).group(1)
NameError: name 're' is not defined
Error occurs
Traceback (most recent call last):
  File "/tmp/pdm-test/.venv/lib/python3.9/site-packages/pdm/builders/base.py", line 80, in log_subprocessor
    subprocess.check_call(
  File "/usr/lib/python3.9/subprocess.py", line 373, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['/usr/bin/python', '-u', '/tmp/pdm-test/.venv/lib/python3.9/site-packages/pdm/installers/_editable_install.py', '…/Vcs/mkdocs/setup.py', '/tmp/pdm-test/__pypackages__/3.9', '/tmp/pdm-test/__pypackages__/3.9/lib', '/tmp/pdm-test/__pypackages__/3.9/bin']' returned non-zero exit status 1.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/tmp/pdm-test/.venv/lib/python3.9/site-packages/pdm/termui.py", line 189, in logging
    yield logger
  File "/tmp/pdm-test/.venv/lib/python3.9/site-packages/pdm/installers/synchronizers.py", line 305, in synchronize
    handlers[kind](key)
  File "/tmp/pdm-test/.venv/lib/python3.9/site-packages/pdm/installers/synchronizers.py", line 148, in install_candidate
    installer.install(can)
  File "/tmp/pdm-test/.venv/lib/python3.9/site-packages/pdm/installers/installers.py", line 45, in install
    self.install_editable(candidate.ireq)
  File "/tmp/pdm-test/.venv/lib/python3.9/site-packages/pdm/installers/installers.py", line 78, in install_editable
    builder.subprocess_runner(install_args, ireq.unpacked_source_directory)
  File "/tmp/pdm-test/.venv/lib/python3.9/site-packages/pdm/builders/base.py", line 197, in subprocess_runner
    return log_subprocessor(cmd, cwd, extra_environ=env)
  File "/tmp/pdm-test/.venv/lib/python3.9/site-packages/pdm/builders/base.py", line 88, in log_subprocessor
    raise BuildError(f"Call command {cmd} return non-zero status.")
pdm.exceptions.BuildError: Call command ['/usr/bin/python', '-u', '/tmp/pdm-test/.venv/lib/python3.9/site-packages/pdm/installers/_editable_install.py', '…/Vcs/mkdocs/setup.py', '/tmp/pdm-test/__pypackages__/3.9', '/tmp/pdm-test/__pypackages__/3.9/lib', '/tmp/pdm-test/__pypackages__/3.9/bin'] return non-zero status.
```

**Comment by frostming** [comment]
Can you help to verify if this issue is fixed by #419 ?

**Comment by blueyed** [comment]
> Can you help to verify if this issue is fixed by #419 ?

Yes, tested it (cc04428), and it fixed the issue.

**Comment by frostming** [comment]
Thanks
